src/training/ea/lipizzaner_gan_trainer.py

```diff
@@ -82,6 +73,27 @@ def __init__(self, dataloader, network_factory, population_size=10, tournament_s
             # TODO: Add code for safe implementation & error handling
             raise KeyError("Fitness section must be defined in configuration file")
 
+        if 'score' in self.settings and self.settings['score'].get('enabled', calc_mixture):
+            self.score_calc = ScoreCalculatorFactory.create()
+            self.score_sample_size = self.settings['score'].get('sample_size', score_sample_size)
+            self.score = float('inf') if self.score_calc.is_reversed else float('-inf')
+            self.mixture_generator_samples_mode = self.cc.settings['trainer']['mixture_generator_samples_mode']
+        else:
+            self.score_sample_size = score_sample_size
+            self.score_calc = None
+            self.score = 0
+
+        if 'optimize_mixture' in self.settings:
+            self.optimize_weights_at_the_end = True
+            self.score_sample_size = self.settings['optimize_mixture'].get('sample_size', es_score_sample_size)
+            self.es_generations = self.settings['optimize_mixture'].get('es_generations', es_generations)
+            self.es_random_init = self.settings['optimize_mixture'].get('es_random_init', es_random_init)
+            self.mixture_sigma = self.settings['optimize_mixture'].get('mixture_sigma', mixture_sigma)
+            self.mixture_generator_samples_mode = self.cc.settings['trainer']['mixture_generator_samples_mode']
+            if not ('score' in self.settings and self.settings['score'].get('enabled', calc_mixture)):
```

Comment:
Can this be written above as an `elif` to make the flow simpler?

Reply:
Sure!